default.nix

```diff
@@ -11,6 +11,13 @@ let
   });
   socket-io-src = pkgs.fetchgit (removeAttrs (lib.importJSON ./pkgs/engine-io.json) ["date"]);
   cardano-sl-src = pkgs.fetchgit (removeAttrs (lib.importJSON ./pkgs/cardano-sl.json) ["date"]);
+  cores   = 4;
+  threads = 8;
+  buildSubset = subset: drv: overrideCabal drv (drv: {
+    buildTarget = subset;
+  });
+  buildSpeedupFlagsMarlowScaling = drv : appendConfigureFlag drv
+    "--ghc-option=-rtsopts --ghc-option=+RTS --ghc-option=-N${toString threads} --ghc-option=-A128m --ghc-option=-n2m --ghc-option=-qb0 --ghc-option=-qn${toString cores} --ghc-option=-RTS";
```

Comment:
1. The problem here is.. we need to set both `-N` and `-qn`, in the `2:1` ratio, as per Simon Marlow's suggestion (see https://ghc.haskell.org/trac/ghc/ticket/9221#comment:72).
2. Once we accept №1, a need to have an authoritative _Nix-evaluation-time_ source of information on the amount of cores on the system.
3. I couldn't find such a source, hence the hard-coding.
4. Since we hard-code `-qn`, I guess it's kinda debatable if leaving `-N` _free_ is a good thing.  It would certainly somewhat obscure the _Marlow ratio_ between cores and threads.